### ISSUE TITLE
A question about the code in trainer.py which may violate the settings in Continual Learning? 

### ISSUE THREAD

**Issue by MoonJian** [hugoycj/Incremental-Learning-with-Weight-Aligning#2]
Thanks for your fantastic work, I have learned a lot from it! But I still have a question about the code in trainer.py, line 94 and 98: 
`train_xs.extend(train_x)`
`train_data = DataLoader(BatchData(train_xs, train_ys, self.input_transform),
                        batch_size=batch_size, shuffle=True, drop_last=True)`
it seems that all of the training data of previous tasks will be rehearsed in current step, which is obviously violating the settings in class-incremental learning...
So could you help me solve my question?
Thanks~

**Comment by SunWenJu123** [comment]
I have the same confusion ...

**Comment by CapableCat** [comment]
I also feel confused, the original is the herding algorithm. And, this project fixed the test set size to 2000.

**Comment by zmhhmz** [comment]
This is a huge problem. I hope the author can explain this...

**Comment by TOM-tym** [comment]
Same issue here. The most surprising thing here is that, although this is not the official implementation, it still claims that it can achieve the results in the paper with such a `BUG'.
